query.go

```diff
@@ -400,7 +400,11 @@ func queryChainServiceBatch(
 					newLogClosure(func() string {
 						return spew.Sdump(queryMsgs[handleQuery])
 					}))
-			case <-matchSignal:
+			case _, ok := <-matchSignal:
+				if !ok {
+					return
```

Comment:
Shouldn't we be resetting the state of this query to `queryWaitSubmit` so that it can be assigned to a different peer? This might also apply to some of other channel reads within this `select` block.

Reply:
Good catch, that seems to be the case! Fixed it for the affected cases.